docs/coordinator/api/remote.md

```diff
@@ -0,0 +1,68 @@
+# API
+
+**Please note:** This documentation is a work in progress and more detail is required.
+
+## Remote Write
+
+Write a Prometheus Remote write query to M3.
+
+### URL
+
+`/api/v1/prom/remote/write`
+
+### Method
+
+`POST`
+
+### URL Params
+
+None.
+
+### Header Params
+
+#### Optional
+
+--8<--
+docs/common/headers_optional_read_write.md
+--8<--
+
+### Data Params
+
+Binary [snappy compressed](https://en.wikipedia.org/wiki/Snappy_(compression)) Prometheus [WriteRequest protobuf message](https://github.com/prometheus/prometheus/blob/10444e8b1dc69ffcddab93f09ba8dfa6a4a2fddb/prompb/remote.proto#L22).
+
+### Sample Call
+
+There is no trivial way to Snappy compress and marshal a Prometheus WriteRequest protobuf message using just shell, so this example uses a specific command line utility instead. 
+
+This sample call is made using `promremotecli` which is a command line tool that uses a [Go client](https://github.com/m3db/prometheus_remote_client_golang) to Prometheus Remote endpoints. For more information visit the [GitHub repository](https://github.com/m3db/prometheus_remote_client_golang).
+
+There is also a [Java client](https://github.com/m3dbx/prometheus_remote_client_java) that can be used to make requests to the endpoint.
+
+Each `-t` parameter specifies a label (dimension) to add to the metric.
+
+The `-h` parameter can be used as many times as necessary to add headers to the outgoing request in the form of "Header-Name: HeaderValue".
+
+Here is an example of writing the datapoint at the current unix timestamp with value 123.456:
+
+<!-- 
+Note: keep this example similar to the one found in query API 
+documentation for consistency/ease of readers.
+-->
+```bash
+docker run -it --rm                                            \
+  quay.io/m3db/prometheus_remote_client_golang:latest          \
+  -u http://host.docker.internal:7201/api/v1/prom/remote/write \
+  -t __name__:http_requests_total                              \
+  -t code:200                                                  \
+  -t handler:graph                                             \
+  -t method:get                                                \
+  -d $(date +"%s"),123.456
+promremotecli_log 2019/06/25 04:13:56 writing datapoint [2019-06-25 04:13:55 +0000 UTC 123.456]
+promremotecli_log 2019/06/25 04:13:56 labelled [[__name__ http_requests_total] [code 200] [handler graph] [method get]]
+promremotecli_log 2019/06/25 04:13:56 writing to http://host.docker.internal:7201/api/v1/prom/remote/write
+{"success":true,"statusCode":200}
+promremotecli_log 2019/06/25 04:13:56 write success
```

Comment:
Should you provide a link to the doc that will explain how they can retrieve the data they just wrote?

Reply:
I'll point them to the query API docs, good idea.